src/Chainweb/Mempool/InMem.hs

```diff
@@ -342,3 +383,9 @@ getRecentTxs maxNumRecent oldHw rlog
     txs = V.map ssnd $ V.takeWhile pred $ _rlRecent rlog
     pred (T2 x _) = x >= oldHw
 
+------------------------------------------------------------------------------
+getMempoolStats :: InMemoryMempool t -> IO MempoolStats
+getMempoolStats m = do
+    withMVar (_inmemDataLock m) $ \d -> MempoolStats
+        <$> (length <$> readIORef (_inmemPending d))
```

Comment:
Prefer `PSQ.size` to `length` here (since `length` probably goes via `toList` and will create garbage)

Use `<$!>`

Reply:
It seems that `PSQ.size` also uses the `Foldable` instance :-(. But using `size` seems better.

Reply:
Ahh! That's disappointing